aten/src/ATen/mps/EmptyTensor.cpp

```diff
@@ -63,6 +64,9 @@ TensorBase empty_mps(
 
     auto memory_format = memory_format_opt.value_or(MemoryFormat::Contiguous);
     tensor.unsafeGetTensorImpl()->empty_tensor_restride(memory_format);
+    if (at::globalContext().deterministicAlgorithms()) {
```

Comment:
Perhaps we want to mark these with `C10_UNLIKELY` to avoid regress performance in the hot path?